app/views/admin/books/_form.html.erb

```diff
@@ -0,0 +1,73 @@
+<%= form_for [:admin, @product], html: {class: "row g-3"} do |f| %>
+  <%= render "shared/error_messages", object: f.object %>
+  <div class="col-md-6">
+    <%= f.label :name, t("input_name_label") %>
+    <span class="text-danger">(*)<span>
+    <%= f.text_field :name,
+        {class: "form-control", placeholder: t("input_name")} %>
+  </div>
+  <div class="col-md-6">
+    <%= f.label :price, t("input_price_label") %>
+    <span class="text-danger">(*)<span>
+    <%= f.number_field :price,
+        {class: "form-control",
+         placeholder: t("input_price"),
+         min: 0, max: 100000 } %>
+  </div>
+  <div class="col-md-6">
+    <%= f.label :category_id, t("category") %>
+    <span class="text-danger">(*)<span>
+    <%= f.select :category_id,
+        options_for_select(@category),
+                          {selected: @category.first},
+                          {class: "form form-control"} %>
+  </div>
+  <div class="col-md-6">
+    <%= f.label :publisher_id, t("publisher") %>
+    <span class="text-danger">(*)<span>
+    <%= f.select :publisher_id,
+        options_for_select(@publishers),
+                          {selected: @publishers.first},
+                          {class: "form form-control"} %>
+  </div>
+  <div class="row g-3">
+    <%= f.fields_for :book_details do |ff| %>
+      <%= render partial: "book_detail_fields", locals: {f: ff} %>
+    <% end %>
+    <div class="col-md-12">
+      <%= link_to_add_association f, :book_details do %>
+        <label class="btn btn-outline-primary">
+          <i class="fas fa-folder-plus"></i>
+          <%= t ".add_book_detail" %>
+        </label>
+      <% end %>
+    </div>
+  </div>
+  <div class="col-md-6">
+    <%= f.label :publish_year, t("publish_year") %>
+    <%= f.date_field :publish_year, {class: "form-control"} %>
+  </div>
+  <div class="col-md-6">
+    <%= f.label :book_authors, t("book_authors") %>
+    <%= f.select :author_ids,
+        options_for_select(@authors),
+                           {include_blank: true},
+                           multiple: true,
+                           class: "form-control select2" %>
+  </div>
+  <div class="col-md-12">
+    <%= f.label :description, t("description") %>
+    <span class="text-danger">(*)<span>
```

Comment:
https://stackoverflow.com/questions/11197671/use-css-to-automatically-add-required-field-asterisk-to-form-inputs